lab_3/main.py

```diff
@@ -234,11 +387,34 @@ def open(self, file_name: str) -> int:
         :param file_name: name of the json file with .json format
         :return: 0 if profile is opened, 1 if any errors occurred
         """
-        pass
+        if not isinstance(file_name, str):
+            return 1
+        with open(file_name, 'r', encoding="UTF-8") as file:
+            profile_dict = json.load(file)
+        self.language = profile_dict['name']
+        self.n_words = profile_dict['n_words']
+        freq_dict = profile_dict['freq']
+
+        size_of_gram = ''
```

Comment:
Why do you define this as a string and later compare it with `len`?

Reply:
Ok, I fixed this.
Now the parameter size_of_gram = 0